nsv13/code/controllers/subsystem/starsystem.dm

```diff
@@ -0,0 +1,199 @@
+	cycle_bounty_timer()
+	if(bounty_pool == 0) //No need to spam when there is no cashola payout
+		return
+	minor_announce("Bounty Payment Processed", "Naval Command") //Temp - replace words and methods
+	var/split_bounty = bounty_pool / 2 //Split between our two accounts
+	var/datum/bank_account/D = SSeconomy.get_dep_account(ACCOUNT_CAR)
+	D.adjust_money(split_bounty)
+	var/datum/bank_account/DD = SSeconomy.get_dep_account(ACCOUNT_MUN)
+	DD.adjust_money(split_bounty)
+	bounty_pool = 0
+
+/datum/controller/subsystem/starsystem/proc/cycle_bounty_timer()
+	addtimer(CALLBACK(src, .proc/bounty_payout), 15 MINUTES) //Cycle bounty payments every 15 minutes
+
+//////NEW GAMEPLAY LOOP///////
+
+/datum/controller/subsystem/starsystem/proc/cycle_gameplay_loop()
+	addtimer(CALLBACK(src, .proc/gameplay_loop), rand(10 MINUTES, 15 MINUTES))
+
+/datum/controller/subsystem/starsystem/proc/gameplay_loop() //A very simple way of having a gameplay loop. Every couple of minutes, the Syndicate appear in a system, the ship has to destroy them.
+	var/datum/starsystem/current_system //Dont spawn enemies where theyre currently at
+	for(var/obj/structure/overmap/OM in GLOB.overmap_objects) //The ship doesnt start with a system assigned by default
+		if(!OM.main_overmap)
+			continue
+		OM.current_system = find_system(OM) //The ship doesnt start with a system assigned by default
+		current_system = OM?.current_system
+	for(var/datum/starsystem/starsys in systems)
+		if(starsys != current_system)
+			starsys.mission_sector = TRUE //set this sector to be the active mission
+			starsys.spawn_asteroids() //refresh asteroids in the system
+			for(var/i = 0, i < starsys.difficulty_budget, i++) //number of enemies is set via the starsystem vars
+				var/enemy_type = pick(subtypesof(/obj/structure/overmap/syndicate)) //Spawn a random set of enemies.
+				modular_spawn_enemies(enemy_type, starsys)
+			priority_announce("Attention all ships, set condition 1 throughout the fleet. Syndicate incursion detected in: [starsys]. All ships must repel the invasion.", "Naval Command")
+
+//////STARSYSTEM DATUM///////
+
+/datum/starsystem
+	var/name = "Hyperspace"
+	var/parallax_property = null //If you want things to appear in the background when you jump to this system, do this.
+	var/level_trait = ZTRAIT_HYPERSPACE //The Ztrait of the zlevel that this system leads to
+	var/visitable = FALSE //Can you directly travel to this system? (You shouldnt be able to jump directly into hyperspace)
+	var/list/enemies_in_system = list() //For mission completion.
+	var/reward = 5000 //Small cash bonus when you clear a system, allows you to buy more ammo
+	var/difficulty_budget = 2
+	var/list/asteroids = list() //Keep track of how many asteroids are in system. Don't want to spam the system full of them
+	var/mission_sector = FALSE
+/datum/starsystem/New()
+	. = ..()
+	addtimer(CALLBACK(src, .proc/spawn_asteroids), 30 SECONDS)
+
+/datum/starsystem/proc/spawn_asteroids()

```

Comment:
Need something to prevent the asteroids spawning in the FTL overmap z

Reply:
I'm not personally sure on that one. I think asteroid hunting in bluespace is a vibe

Reply:
This one is what I heard might be wanted.